src/java.base/share/classes/jdk/internal/misc/Unsafe.java

```diff
@@ -1078,11 +1081,15 @@ public long objectFieldOffset(Field f) {
     /**
      * Reports the location of the field with a given name in the storage
      * allocation of its class.
+     * <p>
+     * Great care must be taken to ensure the passed field name is validated
+     * to be an instance field.  Do not pass arbitrary field names.
      *
      * @throws NullPointerException if any parameter is {@code null}.
      * @throws InternalError if there is no field named {@code name} declared
      *         in class {@code c}, i.e., if {@code c.getDeclaredField(name)}
-     *         would throw {@code java.lang.NoSuchFieldException}.
+     *         would throw {@code java.lang.NoSuchFieldException}, or if such
+     *         a field is static
```

Comment:
This is actually not the case, the native code only checks whether the field exists, not whether it’s also not `static`:
https://github.com/openjdk/jdk/blob/cbcf401170e0600e48ef74770eaa47c84c7e50b0/src/hotspot/share/prims/unsafe.cpp#L492-L503

```suggestion
     *         would throw {@code java.lang.NoSuchFieldException}.
```

Reply:
Hmm, didn't know that field stream included static fields